### PR TITLE
Skip CI builds for documentation changes

### LINKED ISSUE #3147: Exclude /docs from normal checks so that it's easier to iterate on .md files
There a lot of steps for code checkins that we don't need to validate for a doc change. @marlenecota was going to look into copying what had been done for the WinUI repo.

### PR DESCRIPTION
**Pull request by acoates-ms**
Ports over the logic in WinUI where we skip the CI builds for *.md files.

Fixes #3147 , #3146 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3155)

### CODE REVIEW

File: .ado/shouldSkipPRBuild.ps1 [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    foreach($file in $files)
+    {
+        Write-Host "Checking '$file'"
+        $ext = [System.IO.Path]::GetExtension($file)
+        $fileIsSkippable = $ext -in $skipExts
+        Write-Host "File '$file' is skippable: '$fileIsSkippable'"
+
+        if(!$fileIsSkippable)
+        {
+            $allFilesAreSkippable = $false
```
Comment (licanhua) [inline .ado/shouldSkipPRBuild.ps1]: early return false or break?

File: .ado/windows-vs-pr.yml [hunk 2]
```diff
@@ -116,7 +127,7 @@ jobs:
           msbuildArchitecture: $(MSBuildArchitecture) # Optional. Options: x86, x64
           platform: $(BuildPlatform) # Optional
           configuration: $(BuildConfiguration) # Optional
-          msbuildArguments: '/p:PreferredToolArchitecture=x64' # Optional
+          msbuildArguments: "/p:PreferredToolArchitecture=x64" # Optional
```
Comment (licanhua) [inline .ado/windows-vs-pr.yml]: any reason to change from `'` to `"`?

File: .ado/windows-vs-pr.yml [hunk 3]
```diff
@@ -23,8 +23,19 @@ variables:
...
+        name: checkPayload
+        displayName: "Check if build is required for this PR"
+        inputs:
+          targetType: filePath
+          filePath: .ado/shouldSkipPRBuild.ps1
+
   - job: UWPPR
     displayName: Universal PR
+    dependsOn: Setup
+    condition: ne( dependencies.Setup.outputs['checkPayload.shouldSkipPRBuild'], 'True' )
```
Comment (licanhua) [inline .ado/windows-vs-pr.yml]: Will it more reasonable to use stages + jobs?

File: vnext/Scripts/Tfs/Start-TestServers.ps1 [hunk 4]
```diff
@@ -33,7 +33,7 @@ Write-Host 'Started WebSocket server'
 if ($Preload) {
 	Write-Host 'Preloading bundles'
 
-	Start-Sleep -Seconds 12
+	Start-Sleep -Seconds 30
```
Comment (JunielKatarn) [inline vnext/Scripts/Tfs/Start-TestServers.ps1]: TODO: I'll make this a configurable argument.

**Pull request merged**